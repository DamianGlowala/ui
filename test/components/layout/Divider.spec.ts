import { describe, it, expect } from 'vitest'
import { UDivider } from '#components'
import type { TypeOf } from 'zod'
import ComponentRender from '../component-render'

describe('Divider', () => {
  it.each([
    ['basic case', {}],
    ['with label', { props: { label: 'Divider label' } }],
    ['with icon', { props: { icon: 'i-heroicons-academic-cap', ui: { icon: { base: 'flex-shrink-0 w-5 h-5' } } } }],
    ['with avatar', { props: { avatar: {Â src: 'https://repository-images.githubusercontent.com/428329515/43fec891-9030-4601-8233-5d45ba5c6013', alt: 'nuxt-ui' }, ui: { avatar: { size: 'md' }} } }],
    ['with orientation', { props: { orientation: 'vertical' } }],
    ['with type', { props: { type: 'dotted' } }],
    ['with default slot', { slots: { default: 'Default slot' } }],
    // @ts-ignore
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof Divider.props>) => {
    const html = await ComponentRender(nameOrHtml, options, UDivider)
    expect(html).toMatchSnapshot()
  })
})
