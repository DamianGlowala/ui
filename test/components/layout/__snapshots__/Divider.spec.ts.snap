// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Divider > renders basic case correctly 1`] = `
"<div class="flex items-center align-center text-center w-full flex-row">
  <div class="flex border-gray-200 dark:border-gray-800 w-full border-t border-solid"></div>
  <!--v-if-->
</div>"
`;

exports[`Divider > renders with default slot correctly 1`] = `
"<div class="flex items-center align-center text-center w-full flex-row">
  <div class="flex border-gray-200 dark:border-gray-800 w-full border-t border-solid"></div>
  <div class="font-medium text-gray-700 dark:text-gray-200 flex mx-3 whitespace-nowrap">Default slot</div>
  <div class="flex border-gray-200 dark:border-gray-800 w-full border-t border-solid"></div>
</div>"
`;

exports[`Divider > renders with icon correctly 1`] = `
"<div class="flex items-center align-center text-center w-full flex-row">
  <div class="flex border-gray-200 dark:border-gray-800 w-full border-t border-solid"></div>
  <div class="font-medium text-gray-700 dark:text-gray-200 flex mx-3 whitespace-nowrap"><span class="i-heroicons-academic-cap flex-shrink-0 w-5 h-5"></span></div>
  <div class="flex border-gray-200 dark:border-gray-800 w-full border-t border-solid"></div>
</div>"
`;

exports[`Divider > renders with label correctly 1`] = `
"<div class="flex items-center align-center text-center w-full flex-row">
  <div class="flex border-gray-200 dark:border-gray-800 w-full border-t border-solid"></div>
  <div class="font-medium text-gray-700 dark:text-gray-200 flex mx-3 whitespace-nowrap"><span class="">Divider label</span></div>
  <div class="flex border-gray-200 dark:border-gray-800 w-full border-t border-solid"></div>
</div>"
`;

exports[`Divider > renders with orientation correctly 1`] = `
"<div class="flex items-center align-center text-center w-full flex-col">
  <div class="flex border-gray-200 dark:border-gray-800 h-full border-s border-solid"></div>
  <!--v-if-->
</div>"
`;

exports[`Divider > renders with type correctly 1`] = `
"<div class="flex items-center align-center text-center w-full flex-row">
  <div class="flex border-gray-200 dark:border-gray-800 w-full border-t border-dotted"></div>
  <!--v-if-->
</div>"
`;
